[TECH] Utiliser la fonction native de génération d'UUID v4.

HEYGUL opened the pull request:
## :unicorn: Problème
Lors de l'installation des dépendances de l'API, un message nous avertit de l'obsolescence de la version d'uuid que nous utilisons.
Nous utilisons effectivement une version relativement ancienne.
En regardant le [README](https://github.com/uuidjs/uuid) de uuid, on se rend compte que nous sommes dans le cas d'utilisation de cette librairie uniquement pour générer des uuid v4.


## :robot: Proposition
Nous suivons la recommandation de uuid et utilisons la fonction `Crytpo.randomUUID`.

## :rainbow: Remarques
Sur la suggestion de @Steph0, on désactive le cache  (`disableEntropyCache: true`) dans les fichiers liés à l'authentification.

## :100: Pour tester
Les tests doivent passer tout vert :-)

HEYGUL committed:
tech(api): replace usage of uuid.v4 by crypto.randomUUID

As stated in uuid readme, generating a uuid v4 is now native thanks to
Crypto.randomUUID function.

HEYGUL committed:
chore: remove now useless uuid package

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr6935.review.pix.fr)
  * [App (.org)](https://app-pr6935.review.pix.org)
  * [Orga (.fr)](https://orga-pr6935.review.pix.fr)
  * [Orga (.org)](https://orga-pr6935.review.pix.org)
  * [Certif (.fr)](https://certif-pr6935.review.pix.fr)
  * [Certif (.org)](https://certif-pr6935.review.pix.org)
  * [1D](https://1d-pr6935.review.pix.fr)
  * [Admin](https://admin-pr6935.review.pix.fr)
  * [API](https://api-pr6935.review.pix.fr/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr6935.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr6935/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr6935/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr6935/environment)

Steph0 commented on api/lib/domain/services/authentication/fwb-oidc-authentication-service.js:
```diff
@@ -1,4 +1,4 @@
-import { v4 as uuidv4 } from 'uuid';
+import { randomUUID as uuidv4 } from 'crypto';
```
Je sais que cela fait un peu + de modifications, mais je me demande si ça ne serait pas mieux d'y aller jusqu'au bout dans le switch de la lib vers le module Node standard et enlever le `as uuidv4`. Dans ce fichier et dans les autres.

Sachant que actuellement, en effet randomUUID génère un uuid v4, mais rien ne dit que dans une version autre de Node JS cela passe automatiquement par défaut à la v5 (rendant l'alias erroné sans faire exprès).

```suggestion
import { randomUUID } from 'crypto';
// Et du coup ligne 47
const uuid = randomUUID();
```

octo-topi replied:
Ca peut se faire dans une nouvelle PR

Steph0 replied:
Quel intérêt de découper ? Cette PR fait 16 lignes, et avec le changement proposé elle devrait pas dépasser les 30 lignes 😅 , et on aura le contexte pour lequel on a enlevé l'alias.

HEYGUL replied:
Je suis en phase avec @Steph0 : je vais faire la modif :-)

Steph0 commented on api/lib/domain/services/authentication/oidc-authentication-service.js:
```diff
@@ -1,7 +1,7 @@
 import jsonwebtoken from 'jsonwebtoken';
 import querystring from 'querystring';
-import { v4 as uuidv4 } from 'uuid';
+import { randomUUID as uuidv4 } from 'crypto';
```
⚠️ Je me demande si, sur des module un peu plus critiques comme celui-ci qui est un module d'authentification, il ne faut pas prendre en compte la nouveauté qui se cache derrière le switch au module standard de Node à savoir 

> By default, to improve performance, Node.js generates and caches enough random data to generate up to 128 random UUIDs. To generate a UUID without using the cache, set disableEntropyCache to true. Default: false.

Dans le sens où je ne vois pas cette notion de cache sur la librairie actuelle et que, pour un backend, un cache de 128 entrées ça va vite.

Mon questionnement est que du coup est-ce que pour éviter des problématiques de collisions il ne faudrait pas sur les usages un peu plus sensibles de uuid, ajouter l'option `disableEntropyCache` et [éviter des risques de sécurité](https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#session-id-entropy) qu'il peut y avoir sur des informations de session.

HEYGUL replied:
Bien vu @Steph0 ! Merci d'avoir aussi bien creuser la doc, je vais prendre en compte ta suggestion.

francois2metz replied:
Je reviens sur ce changement car je regardais https://github.com/1024pix/pix/pull/7241 qui utilise le `disableEntropyCache`. Le commentaire et le lien ne m'aide pas a comprendre pourquoi on ne voudrait pas utiliser le cache. 
A aucun moment la documentation avertit de risque de collision, le cache est la pour aller plus vite quand c'est possible. On aurait un doute sur l'implémentation de ce cache ?

HEYGUL replied:
En relisant la documentation je partage l'analyse de @francois2metz , on peut laisser le `disableEntropyCache` à sa valeur par défaut.

octo-topi reviewed (approved):
Plus de message concernant UUID dans `npm start`

```
5 vulnerabilities (4 moderate, 1 high)

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

```

HEYGUL committed:
refactor: remove uuidv4 aliases

As randomUUID may use newer version of uuid in the future, we prefer the
use of an independant name.

HEYGUL committed:
tech: disable randomUUID entropy cache on authentication service

The session ID must be unpredictable, thus we disable the entropy cache
See https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#session-id-entropy
